shell/platform/linux/fl_binary_messenger.cc

```diff
@@ -21,12 +21,65 @@ struct _FlBinaryMessenger {
 
 G_DEFINE_TYPE(FlBinaryMessenger, fl_binary_messenger, G_TYPE_OBJECT)
 
+struct _FlBinaryMessengerResponseHandle {
+  GObject parent_instance;
+
+  // Messenger sending response on
+  FlBinaryMessenger* messenger;
+
+  // Handle to send the response with. This is cleared to nullptr when it is
+  // used.
+  const FlutterPlatformMessageResponseHandle* response_handle;
+};
+
+G_DEFINE_TYPE(FlBinaryMessengerResponseHandle,
+              fl_binary_messenger_response_handle,
+              G_TYPE_OBJECT)
+
+static void fl_binary_messenger_response_handle_dispose(GObject* object) {
+  FlBinaryMessengerResponseHandle* self =
+      FL_BINARY_MESSENGER_RESPONSE_HANDLE(object);
+
+  // Respond if the user forgot to
```

Comment:
I think this is dangerous; there's no knowing how the other side of an arbitrary message will treat an empty response. For a method channel it'll be "not implemented" which is not necessarily correct and thus may be confusing. For other channels, it could be anything. Someone might have a custom message channel where sending back a null response is invalid, and hit a state that's supposed to be impossible (vs. having their future never return).

We should do the same thing the macOS embedding does, which is log this as an error to help alert the developr, but not send the response.

Reply:
I've logged this using `g_critical` and updated the documentation to indicate that the message needs to be responded to before the last reference to the handle is dropped.